### ISSUE TITLE
Towards new release 0.25.0

### ISSUE THREAD

**Issue by xhanulik** [OpenSC/OpenSC#3017]
Here is the draft release notes for the upcoming release, feel free to adjust or let me know what is missing. Some of the project cards in https://github.com/OpenSC/OpenSC/projects/13 are still _in progress_, so I will adjust the draft eventually.

---
## Security
* [CVE-2023-5992](https://github.com/OpenSC/OpenSC/wiki/CVE-2023-5992): Side-channel leaks while stripping encryption PKCS#1.5 padding in OpenSC (#2948)
* [CVE-2024-1454](https://github.com/OpenSC/OpenSC/wiki/CVE-2024-1454): Potential use-after-free in AuthentIC driver during card enrollment in pkcs15init (#2962)

## General improvements
* Update OpenSSL 1.1.1 to 3.0 in MacOS build (#2930)
* Remove support for old card drivers Akis, GPK, Incrypto34 and Westcos, disable Cyberflex driver (#2885)
* Fix 64b to 32b conversions (#2993)
* Improvements for the p11test (#2991)
* Fix reader initialization without SCardControl (#3007)
* Make RSA PKCS#1 v1.5 depadding constant-time (#2948)
* Add option for disabling PKCS#1 v1.5 depadding (type 01 and 02) on the card (#2975)
* Enable MSI signing via Signpath CI integration for Windows (#2799)
* Fixed various issues reported by OSS-Fuzz and Coverity in drivers, PKCS#11 and PKCS#15 layer

## minidriver
* Fix wrong hash selection (#2932)

## pkcs11-tool
* Simplify printing EC keys parameters (#2960)
* Add option to import GENERIC key (#2955)
* Add support for importing Ed25518/448 keys (#2985)
## IDPrime
* Support uncompressed certificates on IDPrime 940 (#2958)
* Enhance IDPrime logging (#3003)
## D-Trust Signature Cards
* Add support for RSA D-Trust Signature Card 4.1 and 4.4 (#2943)
## EstEID
* Remove expired EstEID 3.* card support (#2950)
## ePass2003
* Allow SW implementation with more SHA2 hashes and ECDSA (#3012)
## SmartCard-HSM
* Fix SELECT APDU command (#2978)
## MyEID
* Update for PKCS#15 profile (#2965)
## Rutoken
* Support for RSA 4096 key algorithm (#3011)

**Comment by xhanulik** [comment]
Regarding the security relevant bugs reported by OSS-Fuzz, there are two issues
- https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=65684
  -  fixed with https://github.com/OpenSC/OpenSC/commit/c3545018d059b4debde33b9f34de719dd41e5531
- https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=64898 
  - fixed with https://github.com/OpenSC/OpenSC/commit/5835f0d4f6c033bd58806d33fa546908d39825c9

but they are both fixing previously reported and fixed fuzzing issues .

From Coverity high impact issues, there are only problems connected to unit tests for PKCS#1 v1.5 depadding, fixed by #3016.

**Comment by frankmorgner** [comment]
Hi, @alt3r-3go , great to hear! We will update the table once we have created a release candidate. When that is done, you can extend the wiki (and the test result page) by making a pull request here https://github.com/OpenSC/Wiki

**Comment by Jakuje** [comment]
> > * https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=64898
> 
> This one seems noteworthy for the upcoming release, because the [6d1fcd9](https://github.com/OpenSC/OpenSC/commit/6d1fcd9cf82c6501089898066656fbe6737f3ced) was part of 0.24.0.

The UAF could happen only, when the `sc_get_challenge()` would return value 0 / SC_SUCCESS, which would get through the condition [`if (!rv) {`](https://github.com/OpenSC/OpenSC/blob/bfe896ae5d6604478d822a83030c165cac5bce9c/src/pkcs15init/pkcs15-authentic.c#L871C3-L871C13), but not through the condition [`if (_ret < 0) {`](https://github.com/OpenSC/OpenSC/blob/bfe896ae5d6604478d822a83030c165cac5bce9c/src/pkcs15init/pkcs15-authentic.c#L876) to return.

So I agree that it would make sense to get the CVE for this (with low priority as it only affects the enrollment).

**Comment by xhanulik** [comment]
> This one seems noteworthy for the upcoming release, because the [6d1fcd9](https://github.com/OpenSC/OpenSC/commit/6d1fcd9cf82c6501089898066656fbe6737f3ced) was part of 0.24.0. However, it can only be triggered by a malicious card and during modification of the card. If we want to allocate a CVE for this, we could use the description of [CVE-2023-40661](https://github.com/advisories/GHSA-2c2j-2pgv-gfgc) as template.

Here is the draft of the CVE:

## Memory use after free in AuthentIC driver when updating token info

The Use After Free vulnerability was identified within the AuthentIC driver in OpenSC packages, particularly in the card enrollment process using pkcs15-init when a user or administrator enrolls or modifies cards. An attacker must have physical access to the computer system to take advantage of this flaw. The attack requires a crafted USB device or smart card to present the system with specially crafted responses to the APDUs, which are considered high complexity and low severity. This manipulation can potentially allow for compromising card management operations during enrollment.

### References

* https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=64898
    * fixed with 5835f0d4f6c033bd58806d33fa546908d39825c9

**Comment by frankmorgner** [comment]
please also pick up the code signing of the Windows installer in the changelog (#2799)

**Comment by xhanulik** [comment]
The release candidate 1 is out now https://github.com/OpenSC/OpenSC/releases/tag/0.25.0-rc1.

We would appreciate further testing of rc1 (https://github.com/OpenSC/OpenSC/wiki/Smart-Card-Release-Testing); results can be added as PR to https://github.com/OpenSC/Wiki or shared as a comment on this issue.
